### PR TITLE
[primTorch] refs: lerp

### PR DESCRIPTION
**Pull request by kshitij12345**
As per title

TODO:
* [x] Add error inputs
* [x] Add reference inputs

### CODE REVIEW

File: aten/src/ATen/native/cpu/zmath.h [hunk 1]
```diff
@@ -8,32 +8,6 @@
...
 namespace at { namespace native {
 inline namespace CPU_CAPABILITY {
-
-template <typename SCALAR_TYPE, typename VALUE_TYPE=SCALAR_TYPE>
-inline VALUE_TYPE zabs (SCALAR_TYPE z) {
-  return z;
-}
-
-template<>
-inline c10::complex<float> zabs <c10::complex<float>> (c10::complex<float> z) {
```
Comment (kshitij12345) [inline aten/src/ATen/native/cpu/zmath.h:18]: instantiation which returns the same dtype for `complex` is not used any-where in the codebase.

other instantiation which returns real number is same as `std::abs`

Have updated relevant call sites with std::abs where it made sense.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -8138,19 +8138,19 @@ def sample_inputs_lerp(op_info, device, dtype, requires_grad, **kwargs):
         # no broadcast
         SampleInput(make_arg((S, S)), args=(make_arg((S, S)), 0.4)),
         # broadcast rhs
-        SampleInput(make_arg((S, S)), args=(make_arg((S,)), 0.4)),
+        SampleInput(make_arg((S, S)), args=(make_arg((S,)), 0.6)),
         # scalar tensor
         SampleInput(make_arg(()), args=(make_arg(()), 0.4)),
         # broadcast rhs scalar-tensor
-        SampleInput(make_arg((S, S)), args=(make_arg(()), 0.4)),
+        SampleInput(make_arg((S, S)), args=(make_arg(()), 0.6)),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:8145]: Check case where `weight` is greater than 0.5. (There are two paths based on weight)

File: aten/src/ATen/native/cpu/zmath.h (hunk 1 above)
Comment (mruberry) [reply aten/src/ATen/native/cpu/zmath.h:18]: Really nice cleanup

**Review by ngimel — APPROVED** [review]

**Review by mruberry — APPROVED** [review]
cool! Ship it!

**Pull request closed without merging**